apps/rush-lib/src/cli/RushCommandLineParser.ts

```diff
@@ -35,23 +35,29 @@ import { GlobalScriptAction } from './scriptActions/GlobalScriptAction';
 import { Telemetry } from '../logic/Telemetry';
 import { AlreadyReportedError } from '../utilities/AlreadyReportedError';
 import { RushGlobalFolder } from '../api/RushGlobalFolder';
+import { NodeJsCompatibility } from '../logic/NodeJsCompatibility';
 
 /**
  * Options for `RushCommandLineParser`.
  */
 export interface IRushCommandLineParserOptions {
-  cwd?: string;   // Defaults to `cwd`
+  cwd: string;   // Defaults to `cwd`
+
+  /**
+   * @internal
+   */
+  alreadyReportedNodeTooNewError: boolean;
 }
 
 export class RushCommandLineParser extends CommandLineParser {
   public telemetry: Telemetry | undefined;
-  public rushConfiguration: RushConfiguration;
   public rushGlobalFolder: RushGlobalFolder;
+  public rushConfiguration: RushConfiguration;
 
   private _debugParameter: CommandLineFlagParameter;
   private _rushOptions: IRushCommandLineParserOptions;
 
-  constructor(options?: IRushCommandLineParserOptions) {
+  constructor(options?: Partial<IRushCommandLineParserOptions>) {
```

Comment:
As far as I know, there's only one line of code that constructs RushCommandLineParser().  Why are the constructor parameters ultra-optional?  Shouldn't every parameter simply be required?

Reply:
That would also eliminate the duplicated `_normalizeOptions()`

Reply:
I'd rather keep this contract mostly the same in this PR. The `Partial` keeps the constructor contract the same while making the internals simpler.